svm/src/transaction_processor.rs

```diff
@@ -292,14 +292,20 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
 
         execution_time.stop();
 
-        const SHRINK_LOADED_PROGRAMS_TO_PERCENTAGE: u8 = 90;
-        self.program_cache
-            .write()
-            .unwrap()
-            .evict_using_2s_random_selection(
-                Percentage::from(SHRINK_LOADED_PROGRAMS_TO_PERCENTAGE),
-                self.slot,
-            );
+        // Skip eviction when there's no chance this particular tx batch has increased the size of
+        // ProgramCache entries. Note that this flag is deliberately defined, so that there's still
+        // at least one other batch, which will evict the program cache, even after the occurrences
+        // of cooperative loading.
+        if programs_loaded_for_tx_batch.borrow().loaded_missing {
```

Comment:
i guess this one could be less straightforward. better ideas are very welcome.

Reply:
Global cache can also grow via `cache.merge(programs_modified_by_tx);` above, not just by loading missing.

Reply:
how about this? d50c11cf1791d27e446459577728e2db4ffd1e30

Reply:
Better, but the number of insertions and evictions can still be unbalanced because it is only a boolean.